pkg/client/client.go

```diff
@@ -76,6 +81,28 @@ func NewDefaultClient() (*Client, error) {
 	return NewClient("")
 }
 
+// NewDefaultClientWithTimeout creates a client with default parameters connecting to UNIX
+// domain socket and waits for cilium-agent availability at most 30 seconds.
+func NewDefaultClientWithTimeout() (*Client, error) {
+	timeout := time.After(ciliumAgentTimeout)
+	var c *Client
+	var err error
+	for {
+		select {
+		case <-timeout:
+			return c, err
```

Comment:
Please return a more descriptive error, otherwise we'll online see `context deadline exceeded`. Let's use

> timeout while connecting to cilium agent. Is the cilium agent running?

Reply:
I agree. Although the `context deadline exceeded` is not the only type of error which might be returned there. Maybe I should include the `timeout while connecting...` message only if `err` is an instance of `DeadlineExceeded` from golang.org/x/net?

Otherwise we may receive errors like:

> Timeout while connecting to cilium agent. Is the cilium agent running? invalid host format

Reply:
Or even better - we should probably even timeout only when `err` is an instance of `DeadlineExceeded`. Otherwise error should be returned immediately.

Reply:
Good point. Including the original error is fine as well if we provide the additional context that is relates to a connect timeout